### PR TITLE
enhance EasyBuildMeta easyblock: auto-enable installing with pip + fix setup.py of easyconfigs package so installation with setuptools >= 61.0 works

### PR DESCRIPTION
**Pull request by boegel**
cfr. https://github.com/easybuilders/easybuild-easyconfigs/pull/15206

probably fixes https://github.com/easybuilders/easybuild/issues/817

definitely fixes problems reported in https://github.com/EESSI/software-layer/issues/191

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _EasyBuild-4.5.0.eb_ 
 * **SUCCESS** _EasyBuild-4.6.1.eb_ 
 * **SUCCESS** _networkx-2.6.3-foss-2021a.eb_ 
 * **SUCCESS** _matplotlib-3.5.2-intel-2022a.eb_ 

Build succeeded for 4 out of 4 (4 easyconfigs in total)
node3116.skitty.os - Linux RHEL 8.4, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/b412cb4aecd667a1eb76ab7c81035bb4 for a full test report.

File: easybuild/easyblocks/e/easybuildmeta.py [hunk 1]
```diff
@@ -66,6 +66,28 @@ def __init__(self, *args, **kwargs):
             # consider setuptools first, in case it is listed as a sources
             self.easybuild_pkgs.insert(0, 'setuptools')
 
+        # opt-in to using pip for recent version of EasyBuild, if:
+        # - EasyBuild is being installed for Python >= 3.6;
+        # - pip is available, and recent enough (>= 21.0);
+        # - use_pip is not specified;
+        pyver = sys.version.split(' ')[0]
+        pip = which('pip')
```
Comment (ocaisa) [inline easybuild/easyblocks/e/easybuildmeta.py:74]: I think this is problematic, it doesn't take into account the potential usage of `EB_PYTHON`/`EB_INSTALLPYTHON` which means that `pip` may not even be in the users path but could still be usable.

File: easybuild/easyblocks/e/easybuildmeta.py [hunk 2]
```diff
@@ -66,6 +66,28 @@ def __init__(self, *args, **kwargs):
...
+        self.log.info("Python version: %s - pip: %s", pyver, pip)
+        if sys.version_info >= (3, 6) and pip and self.cfg['use_pip'] is None:
+            # try to determine pip version, ignore any failures that occur while doing so;
+            # problems may occur due changes in environment ($PYTHONPATH, etc.)
+            pip_version = None
+            try:
+                pip_version = det_pip_version()
+                self.log.info("Found Python v%s + pip: %s", pyver, pip_version)
```
Comment (ocaisa) [inline easybuild/easyblocks/e/easybuildmeta.py:82]: Shouldn't we be checking major/minor version matching between `python` and `pip`?

**Review by ocaisa — CHANGES_REQUESTED** [review]

File: easybuild/easyblocks/e/easybuildmeta.py (hunk 2 above)
Comment (boegel) [reply easybuild/easyblocks/e/easybuildmeta.py:82]: Good point, using `python -m pip` is indeed a better approach.
That should be tackled in the `PythonPackage` easyblock, as you're doing in #2807 :+1:

**Review by ocaisa — APPROVED** [review]
Changes LGTM, couple of tests and we are good to go.

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _EasyBuild-4.6.1.eb_ 
 * **SUCCESS** _EasyBuild-4.5.0.eb_ 
 * **SUCCESS** _EasyBuild-4.0.0.eb_ 
 * **SUCCESS** _networkx-2.6.3-foss-2021a.eb_ 
 * **SUCCESS** _matplotlib-3.5.2-intel-2022a.eb_ 

Build succeeded for 5 out of 5 (5 easyconfigs in total)
node3116.skitty.os - Linux RHEL 8.4, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/7175bb0c2764fbfa3fda51b7d1ea612b for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _EasyBuild-4.7.0.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3702.slaking.os - Linux UNKNOWN UNKNOWN, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz (haswell), Python 3.9.13
See https://gist.github.com/f33229e1079360fa6a3c610f49983066 for a full test report.

**Pull request merged**